csharp/SecureMemory/SecureMemory.Tests/ProtectedMemoryImpl/ProtectedMemorySecretTest.cs

```diff
@@ -55,6 +55,20 @@ private void TestNullConfiguration(IProtectedMemoryAllocator protectedMemoryAllo
             }
         }
 
+        [Fact]
+        private void TestInvalidConfiguration()
+        {
+            var configuration = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string>()
+            {
+                {"secureHeapEngine", "openssl11"}  // Note the missing "required" settings
+            }).Build();
+
+            Debug.WriteLine("ProtectedMemorySecretFactoryTest.TestInvalidConfiguration");
+            using (var factory = new ProtectedMemorySecretFactory(configuration))
+            {
+            }
+        }
```

Comment:
The test as-is belongs in ProtectedMemorySecretFactoryTest.cs... emphasis on "Factory" :)

Reply:
Ugh!